### ISSUE TITLE
Transcript

### ISSUE THREAD

**Issue by David5i6** [guillep/pharo-core#5719]
At **src/Transcript-Core/ThreadSafeTranscript.class.st**

We can replace:

```smalltalk
ThreadSafeTranscript >> stepGlobal [
"	The superclass method Model>>step indicates a convention that might be used to interoperate 
	synchronously with the UI-thread.  However when multiple Transcript windows are open, their 
	PluggableTextMorphs share a single instance, from the global Transcript.  To avoid potential trouble, 
	this method should not be named #step.  
	
	As well, we need this method to execute even when no Transcript windows are open, so the stream 
	continues to be reset periodically, otherwise it would grow indefinitely. So this method is invoked 
	from WorldState>>runStepMethodsIn:. 
"

	"Next three lines required temporarily to initialize instance variables added to existing instance"
	deferredClear ifNil: [ deferredClear := false ].
	deferredEndEntry ifNil: [ deferredEndEntry := false ].
	stepContents ifNil: [ stepContents := '' ].
	
	deferredClear ifTrue: 
	[ 	deferredClear := false.  
		stepContents := ''.
		self changed: #clearText.
	].
	deferredEndEntry ifTrue: 
	[	deferredEndEntry := false.  	
		self critical: 
		[	stepContents := stream contents.
			stream resetContents.	
		].
		self changed: #appendEntry.	
	].
	
]
```
with 

```smalltalk
ThreadSafeTranscript >> stepGlobal [
"	The superclass method Model>>step indicates a convention that might be used to interoperate 
	synchronously with the UI-thread.  However when multiple Transcript windows are open, their 
	PluggableTextMorphs share a single instance, from the global Transcript.  To avoid potential trouble, 
	this method should not be named #step.  
	
	As well, we need this method to execute even when no Transcript windows are open, so the stream 
	continues to be reset periodically, otherwise it would grow indefinitely. So this method is invoked 
	from WorldState>>runStepMethodsIn:. 
"

	"Next three lines required temporarily to initialize instance variables added to existing instance"
	deferredClear ifNil: [ deferredClear := false ].
	deferredEndEntry ifNil: [ deferredEndEntry := false ].
	stepContents ifNil: [ stepContents := '' ].
	
	deferredClear ifTrue: 
	[ 	deferredClear := false.  
		stepContents := ''.
		self changed: #clearText.
	].
	deferredEndEntry ifFalse: [^self]. 
	deferredEndEntry := false.  	
	self critical: 
	[	stepContents := stream contents.
		stream resetContents.	
	].
	self changed: #appendEntry.	
]
```

P.S: I followed guide and forked the repo but, my image hangs after trying almost any operation with the repo.

**Comment by Ducasse** [comment]
Hello thanks.
Now can you explain on discord what you are doing because we are all committing on numerous projects and it works.

**Comment by David5i6** [comment]
I downloaded pharo launcher for Windows from ain page, started the Pharo 9 image and download the pharo repository from command line (seems that auth din't work). when fixed and select to use the version of the repository the whole system frozens. After about 20 minutes, the VM quits with an out of memory error.

Today I downloaded from archives the latests 64bits VM and image. Seems to works like a charm.

I'm going to try to create a branch do this fix and issue a merge request.
